Tribler/community/tunnel/conversion.py

```diff
@@ -58,13 +85,13 @@ def _decode_cell(self, placeholder, offset, data):
 
     def _encode_create(self, message):
         payload = message.payload
-        packet = pack("!IHH20s", payload.circuit_id, len(payload.node_public_key),
-                      len(payload.key), payload.node_id) + payload.node_public_key + payload.key
+        packet = pack("!IHH20s?", payload.circuit_id, len(payload.node_public_key),
+                      len(payload.key), payload.node_id, payload.exit_candidates) + payload.node_public_key + payload.key
```

Comment:
But the exit-node must be connectable, hence we could just send an ip-address to which the n-1 node should connect to. Won't solve the "n-1 node knows he's the n-1 node". But it could be a nice change to the create/extend messages (to be able to specify an ip in addition to specifying a pubkey).